### PR TITLE
control-service: add new deployment tables

### PR DESCRIPTION
**Pull request by mivanov1988**
# Why
As part of the VEP-2272, we have to introduce a process to synchronize the data jobs from the database to the Kubernetes.

The initial implementation depends on the data_job_deployment table for both reading and writing operations. As the deployment process operates asynchronously with unpredictable durations, users may believe that their deployment has been completed while it may not have.
To address this issue, we'll be introducing a new table called "desired_data_job_deployment" and renaming the current one to "actual_data_job_deployment". The read operations will read from "actual_data_job_deployment" while the write operations will write to "desired_data_job_deployment".

# What
We've implemented new deployment tables.

Just so you know, this is just the second phase of implementation. The following enhancements are planned for future PRs:

- Once the new tables have been utilized, the old one will be deleted.
- We will annotate the method DataJobsSynchronizer.synchronizeDataJobs() with @scheduled.
- Improved exception handling will be integrated.
- More tests will be included in subsequent updates.
- ThreadPool configuration will be tuned and exposed through the application.properties.

Testing Done
Integration tests

Signed-off-by: Miroslav Ivanov [miroslavi@vmware.com](mailto:miroslavi@vmware.com)

### CODE REVIEW

**Review by antoniivanov — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/BaseDataJobDeployment.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  private String schedule;
+
+  private Float resourcesCpuRequestCores; // TODO embeded DataJobDeploymentResources
+
+  private Float resourcesCpuLimitCores;
+
+  private Integer resourcesMemoryRequestMi;
+
+  private Integer resourcesMemoryLimitMi;
+
```
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/BaseDataJobDeployment.java:43]: Yeah, that's why I added this comment. Will do.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/ActualDataJobDeployment.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  private String deploymentVersionSha;
+
+  private OffsetDateTime lastDeployedDate;
+
+  public ActualDataJobDeployment() {
+    super();
+  }
+
+  public ActualDataJobDeployment(
```
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/ActualDataJobDeployment.java:28]: There is an issue with inheritance - https://medium.com/@sergio-sanchez/inheritance-with-lombok-annotations-in-java-springboot-f32a2d873fff

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/BaseDataJobDeployment.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoniivanov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/BaseDataJobDeployment.java:43]: Sorry. I missed/skimmed over the comment. 

But I am going to take this as "great minds think alike"

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/repository/ActualJobDeploymentRepository.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.repository;
+
+import com.vmware.taurus.service.model.*;
```
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/repository/ActualJobDeploymentRepository.java:8]: fixed, thanks

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/ActualDataJobDeployment.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/ActualDataJobDeployment.java:28]: If needed I will introduce @Builder

**Review by dakodakov — APPROVED** [review]

**Pull request merged**